### PR TITLE
feat(ras-acc) add welcome email template for checkout registration

### PR DESCRIPTION
**Pull request by chickenn00dle**
### All Submissions:

* [ ] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes https://app.asana.com/0/1207385073694403/1205250932343413/f

This PR is partnered with the following main plugin PR: https://github.com/Automattic/newspack-blocks/pull/1754

This adds a new welcome email template for readers who register via checkout as part of RAS-ACC.

Screenshot 2024-05-31 at 17 32 46

### How to test the changes in this Pull Request:

Before testing, be sure to check out the changes in https://github.com/Automattic/newspack-blocks/pull/1754

- As admin ensure the new welcome email template as well as the receipt template are enabled via Newspack > Reader Revenue > Emails

Screenshot 2024-05-31 at 17 34 28

- In an incognito window as a logged out reader, make a donation to the test site via a donation block
- Complete checkout, creating an account in the process, and confirm you receive the new welcome email after checkout (pictured above)
- Refresh the page, and make another donation logged in as the same reader from the previous step and confirm you receive the receipt email this time. (They are very similar but the receipt email does not have any welcome text and does not contain a button leading to the site)
- As admin, disable the welcome email and ensure the receipt template is still enabled
- Again as a reader, **log out** and make another donation.
- Confirm you do not receive the welcome email and instead receive the receipt email.

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/reader-revenue/class-reader-revenue-emails.php [hunk 1]
```diff
@@ -120,6 +121,35 @@ public static function add_email_configs( $configs ) {
 				]
 			),
 		];
+		$configs[ self::EMAIL_TYPES['WELCOME'] ]      = [
+			'name'                   => self::EMAIL_TYPES['WELCOME'],
+			'label'                  => __( 'Welcome', 'newspack-plugin' ),
+			'description'            => __( "Email sent to the donor after they've registered while donating.", 'newspack-plugin' ),
```
Comment (dkoo) [inline includes/reader-revenue/class-reader-revenue-emails.php]: In my testing, this doesn't only apply to donations, but to any transaction (this is correct according to the scope of the feature request task). 

```suggestion
			'description'            => __( "Email sent to new reader accounts registered during a transaction.", 'newspack-plugin' ),
```

File: includes/reader-revenue/class-reader-revenue-emails.php [hunk 2]
```diff
@@ -120,6 +121,35 @@ public static function add_email_configs( $configs ) {
...
+			'template'               => dirname( NEWSPACK_PLUGIN_FILE ) . '/includes/templates/reader-revenue-emails/welcome.php',
+			'editor_notice'          => __( 'This email will be sent to a reader after they sign up and contribute to your site.', 'newspack-plugin' ),
```
Comment (dkoo) [inline includes/reader-revenue/class-reader-revenue-emails.php]:
```suggestion
			'editor_notice'          => __( 'This email will be sent to readers when they register an account during a transaction.', 'newspack-plugin' ),
```

**Review by dkoo — CHANGES_REQUESTED** [review]
This is working well! I have a few copy suggestions due to the fact that this feature covers more than just donation transactions.

Also, the `*MAGIC_LINK_URL*` placeholder isn't getting populated in the final sent email. We need to fix that auto-replacement as well as include it as a placeholder in the Welcome email config shown in the editor.

 
Screenshot 2024-06-04 at 3 00 19 PM

File: includes/reader-revenue/class-reader-revenue-emails.php (hunk 1 above)
Comment (chickenn00dle) [reply includes/reader-revenue/class-reader-revenue-emails.php]: Good suggestion! Updated in https://github.com/Automattic/newspack-plugin/pull/3150/commits/c58f6399821ce0dcd1262c47d8d609df225e4a42

File: includes/reader-revenue/class-reader-revenue-emails.php (hunk 2 above)
Comment (chickenn00dle) [reply includes/reader-revenue/class-reader-revenue-emails.php]: Also updated in https://github.com/Automattic/newspack-plugin/pull/3150/commits/c58f6399821ce0dcd1262c47d8d609df225e4a42

**Comment by chickenn00dle** [comment]
Thanks for the review @dkoo!

> Also, the *MAGIC_LINK_URL* placeholder isn't getting populated in the final sent email. We need to fix that auto-replacement as well as include it as a placeholder in the Welcome email config shown in the editor.

This was actually copied over from another template and should not be a verification url. I've updated to a generic account link in https://github.com/Automattic/newspack-plugin/pull/3150/commits/c58f6399821ce0dcd1262c47d8d609df225e4a42

**Comment by chickenn00dle** [comment]
~~Related to https://github.com/Automattic/newspack-blocks/pull/1754#issuecomment-2151639952, I added some small changes in https://github.com/Automattic/newspack-plugin/pull/3150/commits/1edca4b99e6b0a57649c23cb2ad8e17bf451176e to deal with a session expired error that appeared for checkout button checkout modal~~Nevermind. Digging further into this looks like the session issue can be addressed solely in the blocks plugin.

**Review by dkoo — APPROVED** [review]

**Pull request merged**